edge-modules/metrics-collector/src/Settings.cs

```diff
@@ -88,6 +90,7 @@ private static Settings Create()
                     configuration.GetValue<UploadTarget>("UploadTarget", UploadTarget.AzureMonitor),
                     configuration.GetValue<bool>("CompressForUpload", true),
                     configuration.GetValue<bool>("TransformForIoTCentral", false),
+                    configuration.GetValue<bool>("AddIdentifyingTags", false),
```

Comment:
Suggest we name this `experimentalFeatures__AddIdentifyingTags`.  It's verbose, but [inspired by what we already do for things like EdgeHub](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-publish-subscribe?view=iotedge-2020-11#prerequisites).

Reply:
Unfortunately this small change introduced a subtle bug that in the end make renders the complete feature unusable.  "__" is internally replaced by ":" - https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#environment-variable-configuration-provider. So while the environment variable could be named "experimentalFeatures__AddIdentifyingTags" the corresponding key to access this value is "experimentalFeatures:AddIdentifyingTags" - will fix that - as it seems no-one has tested this feature after the setting has been renamed 😢

Reply:
New PR that fixes this: https://github.com/Azure/iotedge/pull/6422